### DISCUSSION TITLE
operates w/ not all haptic ICs and motors connected?

### DISCUSSION THREAD

**Discussion by PWPInnovator898** [kriswilk/buzzah#4] [Q&A]
Hello Kris;
Must all 8 haptic ICs AND all 8 motors be connected for vibrations to begin?
When 4 of 8 haptic ICs and 4 motors are connected..., I see only a brief 2 blink red LED sequence every ~6sec... 
See attached.
Thanks, PAC
![image](https://user-images.githubusercontent.com/128418916/237005489-6d8f91d9-9a0c-4cc1-9a80-6f36968923a8.png)

**Comment by kriswilk** [comment]
My code did not react gracefully if any driver was missing...it would just die as you observed. I just pushed a small change that fixes this. Now if any driver is missing upon startup, that port is simply ignored. Please give the new code a try...

**Comment by PWPInnovator898** [comment]
Your new code works!  Even w/ 170 Hz tactors, too.  I assume they (the 170
Hz tactors) are driven at the autoresonance-determined freq?
I'll provide feedback on intensity soon.  First impressions are decent,
though.
thanks! Phillip

On Tue, May 9, 2023 at 11:05 AM Kris Wilk ***@***.***> wrote:

> My code did not react gracefully if any driver was missing...it would just
> die as you observed. I just pushed a small change that fixes this. Now if
> any driver is missing upon startup, that port is simply ignored. Please
> give the new code a try...
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kriswilk/buzzah/discussions/4#discussioncomment-5851752>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A6TYIZH7YKSE3WHL4IFURT3XFJ2MLANCNFSM6AAAAAAX27O6VU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Thanks and best regards,
Phillip Callahan

**Comment by kriswilk** [comment]
The latest code is still strictly using the auto-resonance feature. Glad to hear that the 170 Hz LRAs work, although based on the datasheet that may just be by luck. I will let you know once I add the option to set the driving frequency manually...

**Comment by PWPInnovator898** [comment]
Sounds good, thanks!

On Tue, May 9, 2023 at 11:44 AM Kris Wilk ***@***.***> wrote:

> The latest code is still strictly using the auto-resonance feature. Glad
> to hear that the 170 Hz LRAs work, although based on the datasheet that may
> just be by luck. I will let you know once I add the option to set the
> driving frequency manually...
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kriswilk/buzzah/discussions/4#discussioncomment-5852139>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A6TYIZHHB5TAE5MHVSWSD33XFJ7BNANCNFSM6AAAAAAX27O6VU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Thanks and best regards,
Phillip Callahan
